ci-operator/step-registry/idp/htpasswd/idp-htpasswd-commands.sh

```diff
@@ -105,21 +105,39 @@ function set_users () {
     oc get $TARGET_RESOURCE -o json > "${oauth_file_src}"
     jq $IDP_FIELD' += [{"htpasswd":{"fileData":{"name":"cucushift-htpass-secret"}},"challenge":"true","login":"true","mappingMethod":"claim","name":"cucushift-htpasswd-provider","type":"HTPasswd"}]' "${oauth_file_src}" > "${oauth_file_dst}"
     oc replace -f "${oauth_file_dst}"
+   
 
-    # wait for oauth-openshift to rollout
-    wait_auth=true
-    expected_replicas=$(oc get deployment oauth-openshift -n "$OAUTH_NAMESPACE" -o jsonpath='{.spec.replicas}')
-    while $wait_auth;
+    echo "Wait htpasswd ready up to 10 minutes"
+    auth_ready=false
+    count=0
+    while [[ $count -lt 40 ]]
     do
+        expected_replicas=$(oc get deployment oauth-openshift -n "$OAUTH_NAMESPACE" -o jsonpath='{.spec.replicas}')
+        if [[ $expected_replicas -lt "1" ]];then
+	    #Can not find expected_replicas, continue
+	    #echo "the expected_replicas must large than 0, now it is ${expected_replicas}"
+	    continue
+        fi
+
         available_replicas=$(oc get deployment oauth-openshift -n "$OAUTH_NAMESPACE" -o jsonpath='{.status.availableReplicas}')
         new_gen=$(oc get deployment oauth-openshift -n "$OAUTH_NAMESPACE" -o jsonpath='{.metadata.generation}')
         if [[ $expected_replicas == "$available_replicas" && $((new_gen)) -gt $((gen)) ]]; then
-            wait_auth=false
+            auth_ready=true
+            break
         else
-            sleep 10
-        fi
+            echo "waiting 15s now. elapsed: 15 * $count seconds"
+            sleep 15s
+            (( count++ ))
+	fi
     done
 
+    if [[ $auth_ready == false ]];then
+        echo "Error: the idp-htpasswd is not ready in given time"
```

Comment:
Could you help also add below debugging info that looks straightforward to inspect all old & new pods' status (like Terminating, Pending etc)?
```
        echo "oc get pods -n $OAUTH_NAMESPACE"
        oc get pods -n $OAUTH_NAMESPACE
```